pkg/server/render.go

```diff
@@ -60,9 +61,26 @@ func (ctrl *Controller) renderHandler(w http.ResponseWriter, r *http.Request) {
 		out = &storage.GetOutput{Tree: tree.New()}
 	}
 
-	fs := out.Tree.FlamebearerStruct(p.maxNodes)
-	res := renderResponse(fs, out)
-	ctrl.writeResponseJSON(w, res)
+	switch p.format {
+	case "json":
+		fs := out.Tree.FlamebearerStruct(p.maxNodes)
+		res := renderResponse(fs, out)
+		ctrl.writeResponseJSON(w, res)
+	case "pprof":
+		pprof := out.Tree.PprofStruct(&tree.PprofMetadata{
+			Type: out.SpyName, // TODO
+			Unit: out.Units,
+		})
+		out, err := proto.Marshal(pprof.Pprof())
+		if err == nil {
+			ctrl.writeResponseFile(w, "profile.pprof", out)
+		} else {
+			ctrl.writeInternalServerError(w, err, "")
```

Comment:
It may be good to give a bit more context about the error in the response

```suggestion
			ctrl.writeInternalServerError(w, err, "failed to serialize data")
```